src/inventory_ui.cpp

```diff
@@ -16,88 +16,396 @@
 #include "item_location.h"
 #include "itype.h"
 
-#include <string>
-#include <vector>
+#include <algorithm>
 #include <map>
+#include <numeric>
 #include <sstream>
-#include <algorithm>
+#include <string>
+#include <vector>
 
-bool inventory_entry::operator == ( const inventory_entry &other ) const {
-    return get_category_ptr() == other.get_category_ptr() && location == other.location;
-}
+/** The minimal gap between two cells */
+static const int min_cell_gap = 1;
+/** The minimal gap between two columns */
+static const int min_column_gap = 2;
+/** The gap between two columns when there's enough space, but they are not centered */
+static const int normal_column_gap = 8;
+/** The gap on the edges of the screen */
+static const int screen_border_gap = 1;
+/** The minimal occupancy ratio (see @refer get_columns_occupancy_ratio()) to align columns to the center */
+static const double min_ratio_to_center = 0.65;
+
+static const item_category weapon_held_cat( "WEAPON HELD", _( "WEAPON HELD" ), -200 );
```

Comment:
Does this translation work? Being file scope static means it's created before `main` is called and therefor before the translation system is started, in other words: it won't be translated at all.